src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -1635,4 +974,43 @@ public void triggerInteraction(@NotNull final IInteractionResponseHandler handle
             markDirty();
         }
     }
+
+    @Override
+    public int getJobModifier()
+    {
+        return getCitizenSkillHandler().getJobModifier(this);
+    }
+
+    // --------------------------- Request Handling --------------------------- //
```

Comment:
hm why are we rerouting requests through here now? and some might cause an NPE with no work building

Reply:
Oh, because doing citizenData.getHandler.get seemed a lot of chaining and cluttering elsewhere.

Reply:
```
final IBuilding workBuilding = data.getWorkBuilding();
        if (workBuilding instanceof AbstractBuildingWorker)
        {
```

It shouldn't